servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/RedirectSingle.java

```diff
@@ -184,14 +185,19 @@ public void onSuccess(@Nullable final StreamingHttpResponse response) {
                     LOGGER.trace("Executing redirect to '{}' for request '{}'", location, request);
                 }
 
+                // Consume any payload of the redirect response
+                final Single<StreamingHttpResponse> nextResponse = response.messageBody().ignoreElements()
+                        .concat(redirectSingle.requester.request(newRequest));
+                final RedirectSubscriber redirectSubscriber = new RedirectSubscriber(target, redirectSingle, newRequest,
+                        redirectCount + 1, sequentialCancellable);
                 terminalDelivered = true;   // Mark as "delivered" because we do not own `target` from this point
-                toSource(response.messageBody().ignoreElements()    // Consume any payload of the redirect response
-                        .concat(redirectSingle.requester.request(newRequest)))
-                        .subscribe(new RedirectSubscriber(target, redirectSingle, newRequest, redirectCount + 1,
-                                sequentialCancellable));
+                toSource(nextResponse).subscribe(redirectSubscriber);
             } catch (Throwable cause) {
                 if (!terminalDelivered) {
-                    safeOnError(target, cause);
+                    // Drain response payload body before propagating the cause
```

Comment:
I think I recall asking this previously, but what if the payload body is very large or doesn't complete (malicious client, networking broken, etc.)? Will our timeouts kick-in at this level?

Reply:
It depends on how users configure timeouts for their use cases. If they have it at the response payload body level, then yes. 
We can add more protection as a separate work item and unify all places, bcz this is not the only place that drains payload.

Reply:
followup PR sgtm